### PR TITLE
Use 32-bit method selector instead of function signature

### PR DESCRIPTION
**Pull request by nlordell**
Use 32-bit method selector instead of function signature for method identification in the runtime.

### Test Plan

CI

### CODE REVIEW

**Review by e00E — APPROVED** [review]
I don't see anything wrong but I also don't understand the reason for this change.

**Comment by nlordell** [comment]
@e00E Its for consistency with events. Additionally, when you debug the `Instance` you will see all the event selectors now which might be able to help with debugging web3 calls.

**Review by fleupold — APPROVED** [review]
Isn't it less convenient for debugging to not see the full function signature but only the first bytes of the hash? Or are we still using the clear text name for error propagation?

E.g:

> ERRO [driver::driver::scheduler::system] Batch 5284224 failed with unretryable error: method 'submitSolution(uint32,uint256,address[],uint16[],uint128[],uint128[],uint16[]):(uint256)' failure: contract call reverted with message: Some("Solutions are no longer accepted for this batch")

**Comment by nlordell** [comment]
Errors come from the full method signature. This is just used for the method lookup table. The reason we need to method lookup table is that the `ethabi::Contract` API assumes that there are no duplicate methods and just returns the first method whose name matches the search parameters.

I had originally done this with method signature, but I thought, for consistency with how we do it for events, we should use the hash selector.

Also, this has the added benefit of `println!("{:?}", instance)` dumping all the method selectors.

File: common/src/hash.rs [hunk 1]
```diff
@@ -14,15 +14,18 @@ where
     output
 }
 
+/// A 32-byte hash prefix.
```
Comment (bh2smith) [inline common/src/hash.rs]: The word prefix here suggests to me that this is only the beginning of a 32-byte hash (like the `0x` to indicate it is a hex representation). Is this intended?

Comment (nlordell) [reply common/src/hash.rs]: Yes, it is the first 4 bytes in a 32-byte keccak hash. I'll clarify in the comments.

**Pull request merged**